### PR TITLE
Display attachment attrs when destroying the record

### PR DESCRIPTION
**Pull request by unijewski**
Hey guys! I don't know if you still accept new features or just maintain the gem but there's behaviour that could be slightly modified. 

The default behaviour in AR is to return the entire object that was destroyed, see the screenshot below.

Screenshot 2022-05-17 at 17 35 57

When a model includes Paperclip, unfortunately, its attributes are gone when the destroyed object is returned. It's a bit confusing, especially when tracking object changes. 

Screenshot 2022-05-17 at 17 56 24

This PR introduces a way to keep the attributes being returned after destroying the record. By default, it doesn't change the original behaviour, but after setting `hide_attrs_to_be_destroyed` to `false` within the model, we can see what file data the record had. It works only for the `destroy` callback, so no changes when we update the file/assign `nil`.

### CODE REVIEW

File: lib/paperclip/attachment.rb [hunk 1]
```diff
@@ -33,7 +33,8 @@ def self.default_options
         whiny:                            Paperclip.options[:whiny] || Paperclip.options[:whiny_thumbnails],
         validate_media_type:              true,
         adapter_options:                  { hash_digest: Digest::MD5 },
-        check_validity_before_processing: true
+        check_validity_before_processing: true,
+        hide_attrs_to_be_destroyed:       true
```
[the remark this answers is not in the document]
Comment (unijewski) [reply lib/paperclip/attachment.rb]: Sounds good, changed.

File: lib/paperclip/attachment.rb [hunk 2]
```diff
@@ -613,5 +614,21 @@ def able_to_store_created_at?
     def has_enabled_but_unset_created_at?
       able_to_store_created_at? && !instance_read(:created_at)
     end
+
+    # Checks whether the option to show attributes to be destroyed is enabled and whether a destroy callback has been
+    # invoked when deleting the entire record.
+    def show_attrs_and_destroy_callback_triggered?
+      @options[:return_file_attributes_on_destroy] && instance.instance_variable_get(:@_destroy_callback_already_called)
```
Comment (ssinghi) [inline lib/paperclip/attachment.rb:621]: why do we have the second condition?

File: lib/paperclip/attachment.rb [hunk 3]
```diff
@@ -613,5 +614,21 @@ def able_to_store_created_at?
...
+    end
+
+    # Sets attachment-related attributes to `nil`.
+    def clear_all_attachment_attributes
+      instance_write(:file_name, nil)
+      instance_write(:content_type, nil)
+      instance_write(:file_size, nil)
+      instance_write(:fingerprint, nil)
+      instance_write(:created_at, nil) if has_enabled_but_unset_created_at?
+      instance_write(:updated_at, nil)
```
Comment (ssinghi) [inline lib/paperclip/attachment.rb:631]: Also, any clue why were the attributes being set to `nil`? It seems to have been added during very early days of the project.

https://github.com/kreeti/kt-paperclip/commit/5494e3cfb6f0a85fb41ee3bead39fc269dbfd17e#diff-a00bc7cd3a1cbcb5ea7b3ab1c666d8c9caaba8907c1c1fe9327e41afef3d4dd6R85

File: lib/paperclip/attachment.rb (hunk 2 above)
Comment (unijewski) [reply lib/paperclip/attachment.rb:621]: First, we check whether the option is enabled (for backwards compatibility) and then whether the destroy callback got triggered.

File: lib/paperclip/attachment.rb (hunk 3 above)
Comment (unijewski) [reply lib/paperclip/attachment.rb:631]: Well, I have no idea, to be honest. Maybe we should get rid of it, WDYT?

**Review by ssinghi — APPROVED** [review]

**Comment by ssinghi** [comment]
@unijewski can you please sign your commits, so that the PR can be merged. Thanks!

**Comment by unijewski** [comment]
Hey @ssinghi the commits are signed off now

**Review by ssinghi — APPROVED** [review]

**Pull request merged**